ReactiveCocoa/Swift/Signal.swift

```diff
@@ -889,13 +478,9 @@ extension SignalType where Value: EventType, Error == NoError {
 extension SignalType {
 	/// Injects side effects to be performed upon the specified signal events.
 	@warn_unused_result(message="Did you forget to call `observe` on the signal?")
-	public func on(event event: (Event<Value, Error> -> ())? = nil, failed: (Error -> ())? = nil, completed: (() -> ())? = nil, interrupted: (() -> ())? = nil, terminated: (() -> ())? = nil, disposed: (() -> ())? = nil, next: (Value -> ())? = nil) -> Signal<Value, Error> {
+	public func on(event event: (Event<Value, Error> -> ())? = nil, failed: (Error -> ())? = nil, completed: (() -> ())? = nil, interrupted: (() -> ())? = nil, terminated: (() -> ())? = nil, next: (Value -> ())? = nil) -> Signal<Value, Error> {
 		return Signal { observer in
-			let disposable = CompositeDisposable()
-
-			_ = disposed.map(disposable.addDisposable)
-
-			disposable += signal.observe { receivedEvent in
```

Comment:
Well technically this could be fixed for the operators where it matters, but not worth it without a holistic solution.